irods_testing_environment/test_manager.py

```diff
@@ -88,15 +83,24 @@ def result_string(self):
         return r
 
 
-    def run(self, fail_fast=True, **kwargs):
+    def run(self, fail_fast=True, options=None, **kwargs):
         """Run managed `test_runners` in parallel.
 
         Arguments:
         fail_fast -- if True, the first test to fail ends the run
+        options -- A list of lists of strings representing options to pass to the scripts running tests
         **kwargs -- keyword arguments to be passed to the `test_runner`'s specific `run` method
         """
         import concurrent.futures
 
+        if options is None:
+            options = [str() for _ in range(len(self.test_runners))]
+
+        if type(options) != list or len(options) != len(self.test_runners):
+            raise ValueError('options must be a list of size equal to number of concurrent executors')
```

Comment:
How about - "options must be a list having a size equal ..."

Reply:
Will do